packable/packable/src/packable/map.rs

```diff
@@ -0,0 +1,240 @@
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::DuplicateKey(_) => f.debug_tuple("DuplicateKey").finish(),
+            Self::Key(err) => f.debug_tuple("Key").field(err).finish(),
+            Self::Value(err) => f.debug_tuple("Value").field(err).finish(),
+            Self::Prefix(err) => f.debug_tuple("Prefix").field(err).finish(),
+        }
+    }
+}
+
+#[cfg(feature = "std")]
+impl<K, KE, VE, P> std::error::Error for UnpackMapError<K, KE, VE, P>
+where
+    KE: std::error::Error,
+    VE: std::error::Error,
+    P: std::error::Error,
+{
+}
+
+impl<K, KE, VE, P> From<Infallible> for UnpackMapError<K, KE, VE, P> {
+    fn from(err: Infallible) -> Self {
+        match err {}
+    }
+}
+
+impl<K, KE: fmt::Display, VE: fmt::Display, P: fmt::Display> fmt::Display for UnpackMapError<K, KE, VE, P> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::DuplicateKey(_) => write!(f, "duplicate key in map"),
+            Self::Key(err) => write!(f, "cannot unpack key: {}", err),
+            Self::Value(err) => write!(f, "cannot unpack value: {}", err),
+            Self::Prefix(err) => write!(f, "cannot unpack prefix: {}", err),
+        }
+    }
+}
+
+/// Error type raised when a semantic error occurs while unpacking an ordered map.
+pub enum UnpackOrderedMapError<K, KE, VE, P> {
+    /// A map error.
+    Map(UnpackMapError<K, KE, VE, P>),
+    /// An unordered map.
+    Unordered,
+}
+
+impl<K, KE, VE, P> From<UnpackMapError<K, KE, VE, P>> for UnpackOrderedMapError<K, KE, VE, P> {
+    fn from(value: UnpackMapError<K, KE, VE, P>) -> Self {
+        Self::Map(value)
+    }
+}
+
+impl<K, KE: fmt::Debug, VE: fmt::Debug, P: fmt::Debug> fmt::Debug for UnpackOrderedMapError<K, KE, VE, P> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Map(err) => f.debug_tuple("Map").field(err).finish(),
+            Self::Unordered => f.debug_tuple("Unordered").finish(),
+        }
+    }
+}
+
+#[cfg(feature = "std")]
+impl<K, KE, VE, P> std::error::Error for UnpackOrderedMapError<K, KE, VE, P>
+where
+    KE: std::error::Error,
+    VE: std::error::Error,
+    P: std::error::Error,
+{
+}
+
+impl<K, KE, VE, P> From<Infallible> for UnpackOrderedMapError<K, KE, VE, P> {
+    fn from(err: Infallible) -> Self {
+        match err {}
+    }
+}
+
+impl<K, KE: fmt::Display, VE: fmt::Display, P: fmt::Display> fmt::Display for UnpackOrderedMapError<K, KE, VE, P> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Map(err) => err.fmt(f),
+            Self::Unordered => write!(f, "unordered map"),
+        }
+    }
+}
+
+#[cfg(feature = "usize")]
+impl<K: Packable + Ord + Hash, V: Packable> Packable for HashMap<K, V>
+where
+    V::UnpackVisitor: Borrow<K::UnpackVisitor>,
+{
```

Comment:
Can this move into its own `hashmap` module like you did for the `BTreeMap`?

Reply:
I've remove `btreemap` instead